raiden/api/python.py

```diff
@@ -578,12 +579,45 @@ def set_total_channel_deposit(
             )
             raise InsufficientFunds(msg)
 
+        blockhash = views.state_from_raiden(self.raiden).block_hash
+        token_network_proxy = channel_proxy.token_network
+        safety_deprecation_switch = token_network_proxy.safety_deprecation_switch(
+            block_identifier=blockhash
+        )
+        if safety_deprecation_switch:
+            msg = (
+                "This token_network has been deprecated. "
+                "All channels in this network should be closed and "
+                "the usage of the newly deployed token network contract "
+                "is highly encouraged."
+            )
+            raise TokenNetworkDeprecated(msg)
+
+        network_balance = token_network_proxy.token.balance_of(
+            address=Address(self.address), block_identifier=blockhash
+        )
+
+        token_network_deposit_limit = token_network_proxy.token_network_deposit_limit(
+            block_identifier=blockhash
+        )
+
+        if network_balance + addendum > token_network_deposit_limit:
+            msg = f"Deposit of {addendum} would have " f"exceeded the token network deposit limit."
+            raise DepositOverLimit(msg)
```

Comment:
IMO this is not the right exception anymore, because `DepositOverLimit` is now a precondition error.